src/Http/Controllers/PostController.php

```diff
@@ -38,6 +46,77 @@ class PostController extends BasePostController
      *
      * @return BaseHttpResponse|JsonResource|JsonResponse|RedirectResponse
      */
+    #[
+        Get(
+            path: "/posts",
+            operationId: "postGetAllWithFilter",
+            description: "Get all posts with pagination (10 items per page by default, page 1 by default)
+            
+    This API will get records from the database and return them as a paginated list. 
+    The default number of items per page is 10 and the default page number is 1. You can change these values by passing the `per_page` and `page` query parameters.
+            ",
+            summary: "Get all posts with pagination",
+            security: [['sanctum' => []]],
+            tags: ["Post"],
+            parameters: [
+                new Parameter(
+                    name: 'per_page',
+                    description: 'Number of items per page',
+                    in: 'query',
+                    required: false,
+                    schema: new Schema(type: 'integer', default: 10)
+                ),
+                new Parameter(
+                    name: 'page',
+                    description: 'Page number',
+                    in: 'query',
+                    required: false,
+                    schema: new Schema(type: 'integer', default: 1)
+                ),
+            ],
+            responses: [
+                new Response(
+                    response: 200,
+                    description: "Get posts successfully",
+                    content: new JsonContent(
+                        properties: [
+                            new Property(
+                                property: 'error',
+                                description: 'Error status',
+                                type: 'boolean',
+                                default: false
+                            ),
+                            new Property(
+                                property: 'data',
+                                description: 'Data',
+                                properties: [
+                                    new Property(
+                                        property: 'post',
+                                        ref: PostListResourceSchema::class,
+                                        description: 'Post',
+                                        type: 'object'
+                                    ),
+                                ],
+                                type: 'object'
+                            ),
```

Comment:
@pxthinh please merge and check the new code from the main branch, then, update this response data